### PR TITLE
feat(tests): Add status of all tests of last x jobs

### PR DESCRIPTION
**Pull request by billyvg**
This shows the tests results of all tests over the last x jobs

image

### CODE REVIEW

File: zeus/api/resources/repository_tests_by_job.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+from zeus.models import Build, Repository, TestCase, Job, Source
+
+from .base_repository import BaseRepositoryResource
+from ..schemas import TestCaseSummarySchema
+
+testcases_schema = TestCaseSummarySchema(many=True, strict=True)
+
+
+class RepositoryTestsByJobResource(BaseRepositoryResource):
+    def get(self, repo: Repository):
```
Comment (billyvg) [inline zeus/api/resources/repository_tests_by_job.py]: how do I make this less shitty @dcramer ?

Comment (dcramer) [reply zeus/api/resources/repository_tests_by_job.py]: which part

Comment (billyvg) [reply zeus/api/resources/repository_tests_by_job.py]: All of it

Comment (dcramer) [reply zeus/api/resources/repository_tests_by_job.py]: idk looks fine to me, does it work

File: zeus/api/resources/repository_tests_by_job.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            )
+            .order_by(Build.number.asc())
+            .limit(20)
+        )
+
+        if not builds:
+            current_app.logger.info("no successful builds found for repository")
+            return self.respond([])
+
+        job_list = db.session.query(Job.id).filter(Job.build_id.in_(builds)).limit(30)
```
Comment (dcramer) [inline zeus/api/resources/repository_tests_by_job.py]: oh this will probably not work out well in some scenarios

Comment (dcramer) [reply zeus/api/resources/repository_tests_by_job.py]: do you need full jobs? if you could pair it down the API needs it'd be easy to return e.g. 1000 jobs

Comment (billyvg) [reply zeus/api/resources/repository_tests_by_job.py]: Probably don't need full jobs

File: zeus/api/resources/repository_tests_by_job.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            test_name = results_by_test.get(row.name, {})
+            test_name[str(row.job_id)] = Result(row.result)
+            results_by_test[row.name] = test_name
+
+        jobs = list(map(lambda id: str(id[0]), job_list))
+
+        results = {}
+        for test_name, result_by_test in results_by_test.items():
+            results[test_name] = []
+            for job_id in jobs:
```
Comment (billyvg) [inline zeus/api/resources/repository_tests_by_job.py]: probably will refactor this

File: webapp/pages/RepositoryTestChart.jsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  getEndpoints() {
+    let {repo} = this.context;
+    return [['testList', `/repos/${repo.full_name}/tests-by-build`]];
+  }
+
+  renderBody() {
+    return (
+      <Section>
+        <TestChart testList={this.state.testList} />
+        <Paginator links={this.state.links} {...this.props} />
```
Comment (dcramer) [inline webapp/pages/RepositoryTestChart.jsx]: i think this is ``testList.links``

**Pull request merged**